### PR TITLE
[wallet] remove redundant KeyOriginInfo access, already done in CreateSig

### PR DESCRIPTION
**Pull request by instagibbs**
This redundancy is confusing as it looks like pubkeyhashes are special in some way based on where it's called.

### CODE REVIEW

**Comment by instagibbs** [comment]
See: https://github.com/bitcoin/bitcoin/pull/14678/files#diff-d546634098d8003b6397c8643b74e47dR122

**Comment by MeshCollider** [comment]
LGTM https://github.com/bitcoin/bitcoin/pull/14678/commits/9ae01b994c20c45c79137a37ab8a192c08fc4b34

File: src/script/sign.cpp [hunk 1]
```diff
@@ -64,10 +64,6 @@ static bool GetPubKey(const SigningProvider& provider, SignatureData& sigdata, c
     }
     // Query the underlying provider
     if (provider.GetPubKey(address, pubkey)) {
```
Comment (practicalswift) [inline src/script/sign.cpp]: Use `return provider.GetPubKey(address, pubkey);` instead :-)

Comment (instagibbs) [reply src/script/sign.cpp]: good one, done.

**Comment by instagibbs** [comment]
@laanwj It's the only instance of it, yes.

**Comment by laanwj** [comment]
so this is `SignStep` that calls `GetPubKey` then `CreateSig`?
I see, utACK https://github.com/bitcoin/bitcoin/pull/14678/commits/f7beb95a1f921ed292680fbd2f042b4bac73bf59

**Comment by instagibbs** [comment]
made `sigdata` const to make the intention of function clearer

**Comment by laanwj** [comment]
re-utACK https://github.com/bitcoin/bitcoin/pull/14678/commits/b81a1860561ccbda3cd6913f54e49fdc91c7718c

**Pull request merged**